src/pinns_pde_solve.jl

```diff
@@ -110,7 +135,31 @@ struct PhysicsInformedNN{isinplace,C,T,P,PH,DER,PE,AL,K} <: AbstractPINN{isinpla
         else
             _derivative = derivative
         end
-        new{iip,typeof(chain),typeof(strategy),typeof(initθ),typeof(_phi),typeof(_derivative),typeof(param_estim),typeof(additional_loss),typeof(kwargs)}(chain,strategy,initθ,_phi,_derivative,param_estim,additional_loss,kwargs)
+
+        if typeof(adaptive_loss) <: AbstractAdaptiveLoss 
+            _adaptive_loss = adaptive_loss
+        elseif adaptive_loss == nothing
+            _adaptive_loss = NonAdaptiveLoss{Float64}()
```

Comment:
why not just do `adaptive_loss = NonAdaptiveLoss{Float64}()` above? And why default to `Float64`? This should probably match `eltype(init_params)`.

Reply:
I had mostly been following the style that was already there.  Good suggestion on the `eltype(init_params)`.  I simplified it a bit but if I'm using `eltype(init_params)` to choose the type then it needs to be default initialized after init_params is calculated so I can get the `eltype`.